pinot-controller/src/main/java/org/apache/pinot/controller/recommender/io/InputManager.java

```diff
@@ -203,13 +204,15 @@ private void reorderDimsAndBuildMap()
     String sortedColumn = _overWrittenConfigs.getIndexConfig().getSortedColumn();
     Set<String> invertedIndexColumns = _overWrittenConfigs.getIndexConfig().getInvertedIndexColumns();
     Set<String> rangeIndexColumns = _overWrittenConfigs.getIndexConfig().getRangeIndexColumns();
+    Set<String> jsonIndexColumns = _overWrittenConfigs.getIndexConfig().getJsonIndexColumns();
     Set<String> noDictionaryColumns = _overWrittenConfigs.getIndexConfig().getNoDictionaryColumns();
 
     /*Validate if there's conflict between NoDictionaryColumns and dimNamesWithAnyIndex*/
     Set<String> dimNamesWithAnyIndex = new HashSet<>();
     dimNamesWithAnyIndex.add(sortedColumn);
     dimNamesWithAnyIndex.addAll(invertedIndexColumns);
     dimNamesWithAnyIndex.addAll(rangeIndexColumns);
+    dimNamesWithAnyIndex.addAll(jsonIndexColumns);
```

Comment:
This part is essentially to ensure sortedColumn/invertedIndexColumns/rangeIndexColumns are dictionary encoded. On the contrary, json index must be raw. L215 needs to be removed. Maybe we should rename `dimNamesWithAnyIndex` to `dimNamesWithDictionaryDependentIndex`.

Reply:
https://github.com/apache/pinot/pull/8988